common/lib/dependabot/notices.rb

```diff
@@ -94,22 +94,6 @@ def self.generate_supported_versions_description(supported_versions, support_lat
       "Please upgrade to one of the following versions: #{versions_string}#{later_description}."
     end
 
-    # Generates a support notice for the given package manager.
-    # @param package_manager [PackageManagerBase] The package manager object.
-    # @return [Notice, nil] The generated notice or nil if no notice is applicable.
-    sig do
-      params(
-        package_manager: PackageManagerBase
-      ).returns(T.nilable(Notice))
-    end
-    def self.generate_support_notice(package_manager)
-      deprecation_notice = generate_pm_deprecation_notice(package_manager)
-
-      return deprecation_notice if deprecation_notice
-
-      generate_pm_unsupported_notice(package_manager)
-    end
-
```

Comment:
This was originally created to follow the notice structure when implementing the deprecation feature. However, after implementing the unsupported error, I realized we don’t need a similar structure to the deprecation notice. We already have a specialized error (Dependabot::ToolVersionNotSupported) used in other cases, so we are generalizing that for unsupported versions.

Reference code: https://github.com/dependabot/dependabot-core/blob/main/common/lib/dependabot/errors.rb#L435